lib/utils/deprecated.js

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-console */
+
+/**
+ * Outputs a deprecation message.
+ * Supports silencing the messages based on the environmental variable.
+ * @param {string} message
+ */
+module.exports = function deprecated(message) {
+  const isNode = typeof process !== 'undefined';
+  const shouldSuppressWarnings =
+    isNode && process.env.SUPPRESS_DEPRECATION_WARNINGS;
+  const shouldOutputMessage = isNode ? !shouldSuppressWarnings : true;
+
+  if (shouldOutputMessage) {
+    console.warn(`DEPRECATED: ${message}`);
+  }
```

Comment:
Here the `isNode` check is repeated unnecessarily.
I would simplify all this to:
```js
    const shouldOutputDeprecationWarnings = process === 'undefined' ? true : !process.env.SUPPRESS_DEPRECATION_WARNINGS;

    if (shouldOutputDeprecationWarnings) {
      console.warn(`DEPRECATED: ${message}`);
    }
```
Eventually extracting `process === 'undefined'` to `isNode`.

Reply:
Good one, thanks! Will adjust.

Reply:
I would even settle on this:

```js
const shouldOutputMessage =
  typeof process === 'undefined' ||
  !process.env.SUPPRESS_DEPRECATION_WARNINGS;
```